bigquery_etl/stripe/__init__.py

```diff
@@ -136,29 +149,36 @@ def import_(
         table = f"{table}${date:%Y%m%d}"
     with (TemporaryFile(mode="w+b") if table else sys.stdout.buffer) as file_obj:
         filtered_schema = FilteredSchema(resource)
-        for row in _get_rows(api_key, date, resource):
+        has_rows = False
+        for row in _get_rows(api_key, date, before_date, resource):
             if format_resources or (table and api_key):
                 row = filtered_schema.format_row(row)
             file_obj.write(ujson.dumps(row).encode("UTF-8"))
             file_obj.write(b"\n")
-        if table:
+            has_rows = True
+        if not has_rows:
+            click.echo(f"no {filtered_schema.type}s returned")
+            sys.exit(1)
+        elif table:
             if file_obj.writable():
                 file_obj.seek(0)
             warnings.filterwarnings("ignore", module="google.auth._default")
+            job_config = bigquery.LoadJobConfig(
+                clustering_fields=["created"],
+                ignore_unknown_values=False,
+                schema=filtered_schema.filtered,
+                source_format=bigquery.SourceFormat.NEWLINE_DELIMITED_JSON,
+                time_partitioning=bigquery.TimePartitioning(field="created"),
+                write_disposition=bigquery.WriteDisposition.WRITE_TRUNCATE,
+            )
+            if "$" in table:
+                job_config.schema_update_options = [
+                    bigquery.SchemaUpdateOption.ALLOW_FIELD_ADDITION
+                ]
```

Comment:
`schema_update_options` can only be set with `write_disposition=bigquery.WriteDisposition.WRITE_TRUNCATE` if the destination is a partition. This is because when it's a whole table `WRITE_TRUNCATE` causes the schema to be replaced instead of updated.